### DISCUSSION TITLE
Feature idea: keyboard shortcuts for collapsing and expanding the active container

### DISCUSSION THREAD

**Discussion by metastable-void** [menhera-org/TabArray#319] [Announcements]
We are planning to add new keyboard shorcuts to collapse/expand the currently active container on the active window.

What key combinations best suit your needs? Your input is very appreciated.

**Comment by ahmet-rttr34** [comment]
[e9b7a82](https://github.com/BuilderIO/gpt-crawler/commit/e9b7a82f518f1ffdb9aca826c46aba27f48deee7)

**Reply by metastable-void** [reply]
Sorry, what do you mean?

**Comment by edumerco** [comment]
I know it is a very personal choice, but for me it could be a toggle with Ctrl-C (C for Container), and by default it is not taken by the browser...
What do you think?

By the way, thank you very much for the extension. It is brilliant and works beautifully. :)

**Comment by paulclaude** [comment]
I think this would be a great idea, but personally I would like to be able to have a shortcut for 'focus on this container' (like ctrl-shift 1,2,3... or customisable). This would transform the extension and make it practically automatic, like (or even better) on Chrome.

**Reply by edumerco** [reply]
I'm super ok with that, it is the same result. :)

**Reply by metastable-void** [reply]
Defining dynamic number of shortcuts may be difficult or impossible with current APIs provided by Firefox.

**Comment by edumerco** [comment]
>  Defining dynamic number of shortcuts may be difficult or impossible
>  with current APIs provided by Firefox.

Is it possible to add the "focus on this container" to the general right
click menu?

In this case, it can be readily available for everyone. :)


-- 
eduardo mercovich

 Donde se cruzan tus talentos 
 con las necesidades del mundo, 
 ahí está tu vocación. 
 (Anónimo)

**Reply by paulclaude** [reply]
this could be a solution, but I do not know if it is easy to create a dynamic submenu in the general menu, which updates with all containers created

**Comment by metastable-void** [comment]
In fact, we currently have "Focus to This Container" buttons in sidebar/popup.

**Reply by paulclaude** [reply]
No possibility of being able to activate via script 'Focus to This Container' of a certain group (JS, System Events, menu subitem of 'Container Tab Groups', etc.)?

**Comment by edumerco** [comment]
Hey! That could be... where is it? I can't find it...

**Reply by metastable-void** [reply]
![Screenshot 2024-09-27 at 0 03 48](https://github.com/user-attachments/assets/15c25f19-70da-4618-bdd0-98efa4b32f5c)

This.

**Comment by edumerco** [comment]
Ah, thank you! I thought I'd explored all options but somehow I didn't saw/used that icon. 

That is the idea, to focus on 1 group closing others. The request is to have that function in an easier to reach way. That is -as an example- directly in the alt/right-click page menu. 

Does that sounds doable? 

Thanks a lot for all your work... :)

**Reply by metastable-void** [reply]
adding the feature at right-click context menus is possible, so I'll consider that for next releases.

**Reply by metastable-void** [reply]
#486

**Reply by metastable-void** [reply]
Implemented in https://github.com/menhera-org/TabArray/commit/cf53ac11d9948daddccce740fc1097233899131b.
